### PR TITLE
Feat(vm): EVM native

### PR DESCRIPTION
**Pull request by birchmd**
### Description
Adds a new native to the MoveVM which is an embedded EVM. This will allow us to deploy the L2 contracts that are part of the OP stack without needing to rewrite them. Actually deploying these contracts as part of the genesis process will need to be done in a follow-up PR. Making examples of how to use these contracts for user flows (e.g. withdraw ETH from the L2) will need to be done in another PR after that.

### Changes
- New native included in `create_session`
- New EVM native interface included in our framework bundle

### Testing
- New `test_evm` in `evm_native.rs`

### Notes
There are still quite a few TODOs left as comments. I do not think any of them are blocking for the devnet launch, but we will need to revisit them for testnet and mainnet.

### CODE REVIEW

File: genesis-builder/framework/evm/sources/Evm.move [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    use aptos_framework::primary_fungible_store::ensure_primary_store_exists;
+    use EthToken::eth_token::get_metadata;
+    use std::error;
+    use std::signer;
+
+    /// For now deploying EVM contracts is restricted to an admin account.
+    /// This restriction may be lifted in the future.
+    const ENOT_OWNER: u64 = 1;
+
+    const OWNER: address = @0x1;
```
Comment (bulenttastan) [inline genesis-builder/framework/evm/sources/Evm.move:12]: We can actually define this as a variable in the `Move.toml` and refer to it as `const OWNER: address = @owner;`

File: moved/src/move_execution/evm_native.rs [hunk 2]
```diff
@@ -0,0 +1,862 @@
...
+    smallvec::SmallVec,
+    std::{collections::VecDeque, sync::LazyLock},
+};
+
+const EVM_NATIVE_ADDRESS: AccountAddress = AccountAddress::ONE;
+const EVM_NATIVE_MODULE: &IdentStr = ident_str!("evm");
+const ACCOUNT_STORAGE_LAYOUT: MoveTypeLayout = MoveTypeLayout::U256;
+static CODE_LAYOUT: LazyLock<MoveTypeLayout> =
+    LazyLock::new(|| MoveTypeLayout::Vector(Box::new(MoveTypeLayout::U8)));
+static ACCOUNT_INFO_LAYOUT: LazyLock<MoveTypeLayout> = LazyLock::new(|| {
```
Comment (bulenttastan) [inline moved/src/move_execution/evm_native.rs:47]: I assume this is a struct for `address, value, code`, but it's not obvious. We can add a comment describing this layout.

File: moved/src/move_execution/evm_native.rs [hunk 3]
```diff
@@ -0,0 +1,862 @@
...
+            }
+        };
+        result
+            .add_resource_op(struct_tag, op)
+            .expect("Cannot have duplicate storage index");
+    }
+}
+
+fn account_info_struct_tag(address: &Address) -> StructTag {
+    let name = format!("Info_{}", address.encode_hex());
```
Comment (bulenttastan) [inline moved/src/move_execution/evm_native.rs:215]: We might wanna call this `Account_{}` instead 🤔

File: genesis-builder/framework/evm/sources/Evm.move (hunk 1 above)
Comment (birchmd) [reply genesis-builder/framework/evm/sources/Evm.move]: Fixed in https://github.com/MovedNetwork/op-move/pull/143/commits/d0d84791b2756844ac7beab0314db5e56d56e451

File: moved/src/move_execution/evm_native.rs (hunk 2 above)
Comment (birchmd) [reply moved/src/move_execution/evm_native.rs:61]: Additional documentation added in https://github.com/MovedNetwork/op-move/pull/143/commits/4e08ce5a1f5b63e9f31ca61f93d8b9012234f49a

File: moved/src/move_execution/evm_native.rs (hunk 3 above)
Comment (birchmd) [reply moved/src/move_execution/evm_native.rs]: Changed in https://github.com/MovedNetwork/op-move/pull/143/commits/9cdd48aab42f21335a3b1a48083a716b7efb14fe

File: moved/src/move_execution/evm_native.rs [hunk 4]
```diff
@@ -0,0 +1,862 @@
...
+
+    // Capture changes in native context so that they can be
+    // converted into Move changes when the session is finalized
+    evm_native_ctx.state_changes.push(outcome.state.clone());
+
+    // Commit the changes to the DB so that future Move transactions using
+    // the same session will see them.
+    evm_native_ctx.db.commit(outcome.state);
+
+    let gas_used = EvmGasUsed::new(outcome.result.gas_used());
```
Comment (bulenttastan) [inline moved/src/move_execution/evm_native.rs:452]: We'll probably use how much it costed using Move resources than passing the EVM cost over. But I think this is a good way to start to compare them later to decide what makes more sense.

**Review by bulenttastan — APPROVED** [review]
Fantastic work 👏

**Review by RomanHodulak — APPROVED** [review]
Great work :feelsgood: 

There are things that could be iterated upon, but I the way I see it this PR is supposed to put it out there and then focus on each part individually as a refactor while there is always a working base.

Some points we could investigate:
* Gas pricing
* Logs as move events to plug into the existing move events to eth logs pipeline
* Storage as resources layout

**Pull request merged**